### PR TITLE
fix: rollback failed exception log

### LINKED ISSUE #103: Most queries raise non-fatal `ValueError: Transaction is already committed`
## Expected Behavior
Connection pool does not raise exceptions

## Actual Behavior
Connection pool raises non-fatal exceptions for SELECT/INSERT transactions

## Steps to Reproduce the Problem
```python
from sqlalchemy import Column, Integer, MetaData, select, Text, Table, create_engine

project_id = 'foo'
instance_id = 'bar'
database = 'baz'

spanner_uri = f'spanner:///projects/{project_id}/instances/{instance_id}/databases/{database}'

# Create a table with one `Text`-type column
engine = create_engine(spanner_uri)
metadata = MetaData(bind=engine)
t = Table("TestTable", metadata, Column("test_column", Text, nullable=False))
metadata.create_all(engine)

with engine.begin() as conn:
    results = conn.execute(select([t.c.test_column])).fetchall()

print(results)
```
```shell
aderk@aderk ~/q/pyle (u/aderk/labdb-spanner)> python txn_bug.py 
Exception during reset or similar
Traceback (most recent call last):
  File "/usr/local/google/home/aderk/.virtualenvs/pyle/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 697, in _finalize_fairy
    fairy._reset(pool)
  File "/usr/local/google/home/aderk/.virtualenvs/pyle/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 893, in _reset
    pool._dialect.do_rollback(self)
  File "/usr/local/google/home/aderk/.virtualenvs/pyle/lib/python3.8/site-packages/google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py", line 773, in do_rollback
    dbapi_connection.rollback()
  File "/usr/local/google/home/aderk/.virtualenvs/pyle/lib/python3.8/site-packages/google/cloud/spanner_dbapi/connection.py", line 267, in rollback
    self._transaction.rollback()
  File "/usr/local/google/home/aderk/.virtualenvs/pyle/lib/python3.8/site-packages/google/cloud/spanner_v1/transaction.py", line 113, in rollback
    self._check_state()
  File "/usr/local/google/home/aderk/.virtualenvs/pyle/lib/python3.8/site-packages/google/cloud/spanner_v1/transaction.py", line 68, in _check_state
    raise ValueError("Transaction is already committed")
ValueError: Transaction is already committed
[]
```

## Specifications

- Version: Python 3.8.8 (default, Apr  1 2021, 17:00:06) 
1c9be70f901ca016c357ced66fe258f01b694787
- Platform: Linux

### PR DESCRIPTION
**Pull request by IlyaFaer**
Fixes #103

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 name = "sqlalchemy-spanner"
 description = "SQLAlchemy dialect integrated into Cloud Spanner database"
-dependencies = ["sqlalchemy>=1.1.13, <=1.3.23", "google-cloud-spanner>=3.3.0"]
+dependencies = ["sqlalchemy>=1.1.13, <=1.3.23", "google-cloud-spanner>=3.3.0", "pytz"]
```
Comment (maffoo) [inline setup.py:22]: Is pytz necessary here?

Comment (IlyaFaer) [reply setup.py]: Without it the tests could not pass. Removed.

File: google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py [hunk 2]
```diff
@@ -798,13 +798,14 @@ def do_rollback(self, dbapi_connection):
...
         """
-        if (
-            not isinstance(dbapi_connection, spanner_dbapi.Connection)
-            and dbapi_connection.connection._transaction
-            and (
-                dbapi_connection.connection._transaction.rolled_back
-                or dbapi_connection.connection._transaction.committed
-            )
+        if isinstance(dbapi_connection, spanner_dbapi.Connection):
+            dbapi_connection = dbapi_connection
```
Comment (larkee) [inline google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py:802]: nit: this line is redundant which means this check can be refactored. See:

```
if not isinstance(dbapi_connection, spanner_dbapi.Connection):
            dbapi_connection = dbapi_connection.connection
```

**Review by larkee — APPROVED** [review]
LGTM. Just a minor nit which could be addressed in the future cleanup PR.

File: google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py (hunk 2 above)
Comment (IlyaFaer) [reply google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py:802]: Done

**Pull request merged**